### ISSUE TITLE
Screen Reader Marker Plugin Accessibility

### ISSUE THREAD

**Issue by Leaderfirestar** [mistic100/Photo-Sphere-Viewer#733]
PSV version: 4.2.1
Three-js version: 0.129.0
Chrome version: 104.0.5112.102

Hi! I'm trying to build in some accessibility features for my project, and to test it, I'm using Google's "TalkBack" screen reader. It works fine in the rest of my app, but when I open up a Viewer, I can't tap on markers. I know screen readers like to use drag events for it to know where the user is looking and what they're focused on. It should be noted that I have Markers plugin and Gyroscope plugin enabled. After reading the documentation [here](https://photo-sphere-viewer.js.org/plugins/plugin-markers.html#markers-definition) and looking through all Marker related information [here](https://photo-sphere-viewer.js.org/api/), I couldn't find a way to get it to select the marker so I could "click" it. Am I just missing something? Is this in a newer version? My marker is created like this:

```
const marker = {
	id: element.id,
	longitude: element.longitude,
	latitude: element.latitude,
	image: element.image,
	width: 45,
	height: 50,
	anchor: 'bottom center',
	tooltip: "Click Me!"
};
```

**Comment by mistic100** [comment]
There is no API to programatically select a marker. More over there is no concept of "select a marker". When there is a click event on a marker, the plugin dispatches a "select-marker" event, that's it, nothing more changes visually. You could manually invoke the same function you hooked to this event if you want.

**Comment by Leaderfirestar** [comment]
When I enter the Viewer, if I tap and hold over the back button or toggle gyroscope, it highlights it in a blue border to indicate that the screen reader has focused it. Then if I double tap, it clicks it. I cannot the screen reader to focus on the marker just below the back button.

```
markersPlugin.on('select-marker', (e, marker) => {
		console.log('********************MADE IT TO LINE 132********************');
		setShowModal(true);
	});

	markersPlugin.on('over-marker', (e, marker) => {
		console.log('********************MADE IT TO LINE 139********************');
	});
```
Neither of my logs are being fired on mobile (verified through chrome://inspect/#devices), so I can't use the event handlers to open my modal. Is there anything I can do with the markers to make the screen reader recognize it as an element? I can only click it if the screen reader can find it

![Screenshot_20220824-215302_Chrome](https://user-images.githubusercontent.com/42817142/186565752-5b41bde9-f0d7-4c09-8ff5-a625d4a80732.jpg)

**Comment by mistic100** [comment]
Disclaimer : I have zero knowledge in accessibility. But the markers being all contained in a single SVG element I don't think you could do anything with it.

I don't have the answers for what you ask.

**Comment by mistic100** [comment]
you can replace

```
html: `<img src="${markerIcon}" style='max-width: 45px; max-height: 50px' >`,
```

by

```
image: markerIcon
```

**Comment by Leaderfirestar** [comment]
If I do that, will I still be able to throw on an alt/role? I modified my tag slightly to be

```
html: `<img src="${markerIcon}" style="max-width: 45px; max-height: 50px" alt="${element.title}" role="button">`
```
To inform the screen reader that it's clickable (role) and the alt so the screen reader would read what I wanted vs "marker icon active point *location* graphic"
